torch/fx/passes/backends/cudagraphs.py

```diff
@@ -0,0 +1,53 @@
+import torch
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
+from torch.fx.passes.operator_support import OperatorSupport
+from torch.fx.passes.tools_common import CALLABLE_NODE_OPS
+from torch.fx.passes.fake_tensor_prop import FakeTensorProp
+from torch.utils._pytree import tree_map
+
+import operator
+
+class CudaGraphsSupport(OperatorSupport):
```

Comment:
Ok I think I get what is happening at a high level

1. You check to see if an fx graph works with your CUDA graph and if not then partition it and remove mutation
2. You take your fx graph and then stitch in CUDA graph modules where appropriate. For each CUDA graph you warm it up with by opening a CUDA stream and and "record and replay" (trace?) the operations
3. Finally you register a new AOTAutogradStrategy that does all of the above automatically if you use it within a `with optimize()` scope

Reply:
yep